### DISCUSSION TITLE
templater.py silently failing?

### DISCUSSION THREAD

**Discussion by benjamin-cash** [ufs-community/uwtools#301] [Q&A] [answered]
Hi everyone,

I am running into an odd situation with templater.py where it is silently failing to produce output files, and potentially failing to run at all. 

(uwtools) login4.frontera(1063)$  python /work2/02441/bcash/frontera/workflow-tools/src/uwtools/config/templater.py -i /work2/02441/bcash/frontera/ufs_input/prototype-p8/IC/templates/C384.common_file_list.txt -o test.common_file_list.txt
(uwtools) login4.frontera(1064)$ ls -l test.common_file_list.txt
ls: cannot access test.common_file_list.txt: No such file or directory
(uwtools) login4.frontera(1065)$ python /work2/02441/bcash/frontera/workflow-tools/src/uwtools/config/templater.py -i /work2/02441/bcash/frontera/ufs_input/prototype-p8/IC/templates/C384.common_file_list.txt --values_needed
(uwtools) login4.frontera(1066)$ python /work2/02441/bcash/frontera/workflow-tools/src/uwtools/config/templater.py
(uwtools) login4.frontera(1067)$ 

It's been a while since I have used the templater.py tool, so maybe the commands have changed? From the documentation it seems like the above commands should have produced at least some output, even if just an error message for including the wrong options, so I am confused as to why it is completely silent. Any help here would be much appreciated. :)

**Comment by christinaholtNOAA** [comment, marked as answer]
@benjamin-cash We've had a bit of code re-org lately and moved around our command line entry points. We appreciate you bringing this to our attention so we can prioritize re-working the documentation.

Now you can get the same templater functionality from the older version of running a script by calling the command line tool -- given that you have installed the conda package in your environment.

Give this interface a try for your templater needs:

```
uw template render -h
```

This will now be the pattern for all the command line tools in a single modal interface. All levels of the tool respond nicely to a `-h` flag to tell you your next choices. For example, if you wanted to compare two config files:

```
uw config compare -h
```

Or, if you want to see all the things that you can do with a config file:

```
uw config -h
```

Let me know if this works for you.

**Reply by benjamin-cash** [reply]
That did the trick, thanks!

**Reply by christinaholtNOAA** [reply]
Great! Thanks for the follow up.
